packages/wonder-blocks-core/components/with-ssr-placeholder.js

```diff
@@ -7,19 +7,22 @@ import * as React from "react";
  */
 type Props = {|
     /**
-     * The content that is client-only.
+     * The content that is client-only.  This is what is rendered when
+     * not server-side rendering, or (when server-side rendering) after
+     * the initial rehydration has finished.
      */
     children: () => ?React.Node,
 
     /**
-     * Optionally, instead of rendering nothing the first time around, this
-     * component can render a placeholder. To enable this behavior,
-     * pass a "placeholder" prop.
+     * What to render during server-side rendering, or null not to
+     * render anything.
      *
-     * NOTE: Make sure the placeholder will render the same for both client and
-     * server, or it defeats the purpose of using the NoSSR component.
+     * NOTE: Make sure the placeholder will render the same for both
+     * client and server -- that is, it does the same thing for both
+     * the server-side renderer and the rehydration -- or it defeats
+     * the purpose of using the WithSSRPlaceholder component.
      */
-    placeholder?: () => ?React.Node,
+    placeholder: null | (() => React.Node),
```

Comment:
Why change the return type of the placeholder function? I think it's perfectly valid to let the calling code decide of nothing is the right return value.

The `null` can also be dropped and replaced with `?` like `?(() => ?React.Node)`.

Reply:
Good points!  Will change.